src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -392,6 +406,26 @@ def decode(self, embed_ind):
         quantize = quantize.permute(0, 2, 1)
         return quantize
 
+    def forward(self, hidden_states):
+        device = hidden_states.device
+
+        hidden_states = hidden_states.permute(0, 2, 1)
+
+        embed_ind, quantize = self.codebook(hidden_states)
+
+        loss = torch.tensor([0.0], device=device, requires_grad=self.training)
+
+        if self.training:
+            # Pass the gradients straight through the quantization, by directly linking the gradients of the input
+            # embed_ind with the output quantize in the computation graph.
+            quantize = embed_ind + (quantize - embed_ind).detach()
+
+            commitment_loss = nn.functional.mse_loss(quantize.detach(), embed_ind)
+            loss = loss + commitment_loss
```

Comment:
Shall we add a config attribute for the commitment weight? Would only have this default to 1 if we are certain a user would not want to override this

What I would do in the init is set:
```python
self.commitment_weight = config.commitment_weight
```
Then in the forward:
```suggestion
            loss = loss + commitment_loss * self.commitment_weight
```

Reply:
Would actually be inclined to compute the loss as much as possible in the top level model (i.e. `EncodecModel`), rather than scattered across the sub-models. This should make the loss formulation easier to follow

Reply:
Good point. I was looking through the code again, and I think it may be worth restructuring to actually compute the commitment loss during encode() rather than forward() for the EncodecEuclideanCodebook and EncodecVectorQuantization. 

This is a little off convention, but I believe it makes more sense since commitment loss depends only on encoding and quantization, but not decoding.

Reply:
Sounds good to me! Think it's nice to partition the losses this way (split between encoder and decoder), but wouldn't go any more partitioned than this to keep the loss computation easy to follow

Reply:
I've updated the code to compute the commitment loss inside encode() rather than forward(), as it is dependent only on the encoding (during the quantization steps). 

I've also pulled it up to the top level, but this comes at the expense of more memory usage as the intermediate steps must now be stored and passed up to the top. 
1. What do you think about this? Do you think the extra clarify here is worth the memory overhead?
2. Should I just compute the loss inside _encode_frame(), so we only incur the memory overhead of the quantization_steps for one frame at a time?

Regarding the weight config, I think we can factor that in when we add the commitment_loss to the total loss (probably somewhere in the balancer step).

Reply:
I was thinking over this last night, and imho it's better to commit the commitment loss from within the quantizer and pass it back up (rather than computing it at the top level). The reasons are:

1. This mirrors the code in FB encodec/audiocraft, which will make it easier to maintain in the future
2. The quantization loss is a special case loss that depends only on the quantizer, and so it is somewhat natural for it to be there
3. Reduces unnecessary memory overhead

There will still be another area of code at the top level where we will add the commit loss into the other losses (adversarial, and spectrogram loss). 

Let me know what you think!

Reply:
Good with me, particularly for cleaner code and bypassing unnecessary complexity with passing the decoded latents around from the quantiser to the main encode method